server/item/recipe/item.go

```diff
@@ -0,0 +1,101 @@
+package recipe
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+)
+
+// Item is a recipe item. It is an item stack inherently, but it also has an extra value for if it applies to all types.
+type Item struct {
+	item.Stack
+	// AllTypes is true if if the item applies to all of it's type. This is so it can be rendered in the recipe book
+	// for alternative types.
+	AllTypes bool
+}
+
+// inputItem is an item that is inputted to a crafting menu.
+type inputItem struct {
+	// Name is the name of the item being inputted.
+	Name string `nbt:"name"`
+	// MetadataValue is the meta of the item. This can change the item almost completely, or act as durability.
+	MetadataValue int32 `nbt:"meta"`
+	// Count is the amount of the item.
+	Count int32 `nbt:"count"`
+}
+
+// ToStack converts an input item to a stack.
+func (i inputItem) ToStack() (Item, bool) {
```

Comment:
This doesn't actually convert it to an item.Stack but an `Item`. Naming is a bit weird there. Also, you should probably make this unexported.